BSD peertracker occasionally panics on shutdown

The BSD peertracker uses kqueue/kevent to track processes by hanging a blocking read on a kqueue file descriptor. When signaling for shutdown, this read can unblock with an EINTR error before the agent has had a chance to handle the signal. A kqueue read error is currently "unrecoverable", so SPIRE panics when this condition is encountered.

The EINTR possibility was originally unforeseen as the set of possible error conditions evaluated included only those documented directly by the kqueue subsystem. Turns out that EINTR is a generally accepted behavior and thus not directly documented by most syscalls.

The panic is triggered only occasionally and has been witnessed on Darwin. Some brief research suggests that Go normally sets SA_RESTART, thereby abstracting away the EINTR behavior, however there are some inconsistencies/subtleties around how this is handled, particularly on Darwin.

Regardless of the SA_RESTART details, the EINTR condition is generally recoverable. Update the BSD peertracker to detect EINTR and skip the panic.

```
panic: unrecoverable error while reading from kqueue: interrupted system call

goroutine 112 [running]:
github.com/spiffe/spire/pkg/common/peertracker.(*bsdTracker).receiveKevents(0xc00029c6f0, 0x7)
        /Users/azdagron/s/spire/pkg/common/peertracker/tracker_bsd.go:128 +0x2f4
created by github.com/spiffe/spire/pkg/common/peertracker.newTracker
        /Users/azdagron/s/spire/pkg/common/peertracker/tracker_bsd.go:40 +0x116
227d860b0025bd23f7262bc9fe2b4023c6d22c2200c34f5eb86d6cd91a9f682e
```

--- a/pkg/common/peertracker/tracker_bsd.go
+++ b/pkg/common/peertracker/tracker_bsd.go
@@ -124,6 +124,10 @@ func (b *bsdTracker) receiveKevents(kqfd int) {
 				return
 			}
 
+			if err == syscall.EINTR {
+				continue
+			}
+
 			panicMsg := fmt.Sprintf("unrecoverable error while reading from kqueue: %v", err)
 			panic(panicMsg)
 		}

